src/Compilers/Core/Portable/InternalUtilities/OneOrMany.cs

```diff
@@ -103,7 +118,7 @@ public bool Contains(T item)
 
         public OneOrMany<T> RemoveAll(T item)
         {
-            if (HasOne)
+            if (HasOneItem)
             {
                 return EqualityComparer<T>.Default.Equals(item, _one) ? default : this;
             }
```

Comment:
i'm genuinely confused by this method. at the very end it says: return builder.Count == Count ? this. Which effectively says "if i'm removing all elements... return this full collection", instead of "if i'm removing all elements, return an empty OneOrMany.

we should be able to show a test where this breaks.

the `return default` on line 137 is also broken afaict.